### PR TITLE
feat: plugin.Setup with structured config

### PR DESCRIPTION
**Pull request by f0rmiga**
This allows plugins to access some metadata from the aspectplugins.

### CODE REVIEW

File: pkg/plugin/sdk/v1alpha3/plugin/interface.go [hunk 1]
```diff
@@ -35,7 +35,38 @@ type Plugin interface {
 		isInteractiveMode bool,
 		promptRunner ioutils.PromptRunner,
 	) error
-	Setup(properties []byte) error
+	Setup(config *SetupConfig) error
+}
+
+// SetupConfig represents a plugin configuration parsed from the aspectplugins
+// file.
+type SetupConfig struct {
```
Comment (f0rmiga) [inline pkg/plugin/sdk/v1alpha3/plugin/interface.go:43]: @alexeagle Is this sufficient?

File: plugins/error-augmentor/plugin.go [hunk 2]
```diff
@@ -45,7 +44,6 @@ func NewDefaultPlugin() *ErrorAugmentorPlugin {
 
 func NewPlugin() *ErrorAugmentorPlugin {
 	return &ErrorAugmentorPlugin{
-		properties:          nil,
```
Comment (f0rmiga) [inline plugins/error-augmentor/plugin.go:48]: @JesseTatasciore This didn't seem to be used. Was this just a left-over?

File: pkg/plugin/sdk/v1alpha3/plugin/interface.go (hunk 1 above)
Comment (alexeagle) [reply pkg/plugin/sdk/v1alpha3/plugin/interface.go:43]: yeah this covers my use case, thanks!

**Review by alexeagle — APPROVED** [review]

File: plugins/error-augmentor/plugin.go (hunk 2 above)
Comment (JesseTatasciore) [reply plugins/error-augmentor/plugin.go:48]: Yeah it must have been

**Pull request merged**